Automattic/newspack-plugin
# feat(ia): added `useWizardApiFetch` hook and `<WizardsActionCard />` & `<WizardsPluginCard />`

### All Submissions:


### Changes proposed in this Pull Request:

#### Redux store:
- Added `getWizardData` selector to allow data retrieval without making a request i.e. `getWizardAPIData`.
- Added `isLocalError` to flag errors needing to be thrown and bubble up to be caught by consuming component/hook.

#### (new) `useWizardApiFetch` hook:
- Provides a way to make API fetch requests using the Wizard API Redux store. It handles caching, error management, request status and state updates efficiently.
- Caches promise and responses to avoid unnecessary API calls

#### `<WizardActionCard />`:
- Houses a Wizard specific abstraction of the `<Card />` component.
- Utilizes `notification` property to display errors specific to that instance of the component.

#### `<WizardPluginCard />`:
- Unifies common card+plugin usages for easier reuse.
- Utilizes `useWizardApiFetch` to make plugin requests.

#### Connections:
- Contains updates relevant to testing i.e. demo purposes.
- Note: all demo content will be removed on future iterations of the IA feature.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets
2. Navigate to `/wp-admin/admin.php?page=newspack-settings#/`
3. Observe the JSON output at the top of the page (beneath the "Connections" title)
![Screenshot 2024-06-03 at 09 21 23](https://github.com/Automattic/newspack-plugin/assets/3676975/d79540d9-65e0-4139-b359-3cd43c4bfe65)
4. Once the request has finished, you should see the below
![np-ia local_wp-admin_admin php_page=newspack-settings (4)](https://github.com/Automattic/newspack-plugin/assets/3676975/f29f186b-8573-412d-8ec9-90b403a0084c)
5. Observations (confirm all):
   - We have forced an error when hitting "/newspack/v1/plugins/google-site-kit**s**", this should be present under `error` property. 
   - Confirm `statusTwo` property value is `"Error"` too, this represents the outcome of the forced error.
   - When request runs, ensure `isFetching` state updates according to the async fetch cycle of the request.
   - Confirm when switching tabs that the cache/state is not purged and all info (including errors) is retained between transitions

### Other information:
